### DISCUSSION TITLE
How to integerate this in Cmake based C++ project

### DISCUSSION THREAD

**Discussion by anuragvohraec** [uNetworking/uWebSockets#1539] [Q&A]
Hi ,

I have tried various ways to integerate this into my  Cmake based C++ project , without a sucess.

Steps followed:
1. I git cloned the project: `git clone --recurse-submodules git@github.com:uNetworking/uWebSockets.git`
2. I wanted to use the `boringssl` with `libuv`.
3. And hence first I build the boring ssl :
```
mkdir build
cd build
cmake ../
make
```
4. Then I made uSockets: `WITH_BORINGSSL=1 WITH_LIBUV=1 make`
5. And then tried building `uWebSockets`, by command `WITH_BORINGSSL=1 WITH_LIBUV=1 make -f ./GNUmakefile`

With gcc 10.4.0 its was not compiling, so I repeated the process with gcc: 9.4.0 (Built ssl > usockets > uWebSockets).
After which it successfully was build : `./HelloWorld` example is running

I included the header files in my Cmake, and referenced the `uSockets.a` in my project. But still getting many errors.

Can any one summarize or give me link to how to embed `uWebSockets` to a C++ based project with boringssl and libuv event loop ?

My system specfications are:
OS: Ubuntu 20.04.5 LTS
GCC: 10.4.0

**Comment by uNetworkingAB** [comment]
You literally hit "make" in the folder. That's it. It builds all the examples including uSockets.

It also literally prints the exact compiler invocation like so:

`g++ -O3 -Wpedantic -Wall -Wextra -Wsign-conversion -Wconversion -std=c++2a -Isrc -IuSockets/src -flto examples/EchoServer.cpp  uSockets/*.o -lz -o EchoServer`

Which means that it becomes extremely easy and obvious what you need to replicate in <insert one of the 59 million different build systems for C++>. It is very rare for a project to be this clean in its invocation of the compiler, making it incredibly easy to understand and move into whatever environment or IDE or project file you may want.

**Reply by anuragvohraec** [reply]
OK managed to make it finally run. Thank you.

**Comment by uNetworkingAB** [comment]
```
lto1: internal compiler error: compressed stream: data error

0x7fe3c24c8082 __libc_start_main

        ../csu/libc-start.c:308

Please submit a full bug report,

with preprocessed source if appropriate.

Please include the complete backtrace with any bug report.

See <file:///usr/share/doc/gcc-9/README.Bugs> for instructions.                                                                                                                              

lto-wrapper: fatal error: g++ returned 1 exit status                                                                                                                                         

compilation terminated.
```

These are not uWS bugs, your compiler is broken. Report bugs to your compiler vendor, not here.
